pinot-plugins/pinot-file-system/pinot-hdfs/src/main/java/org/apache/pinot/plugin/filesystem/HadoopPinotFS.java

```diff
@@ -202,17 +202,20 @@ public void copyFromLocalFile(File srcFile, URI dstUri)
 
   @Override
   public boolean isDirectory(URI uri) {
-    FileStatus fileStatus = new FileStatus();
-    fileStatus.setPath(new Path(uri));
-    return fileStatus.isDirectory();
+    try {
+      return _hadoopFS.isDirectory(new Path(uri));
```

Comment:
hadoop's source code saying `True iff the named path is a directory. Note: Avoid using this method. Instead reuse the FileStatus returned by getFileStatus() or listStatus() methods.` Try using `_hadoopFS.getFileStatus(new Path(uri)).isDirectory();`

Reply:
Thanks for pointing it out!